sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -27,6 +27,7 @@ public partial class TextAnalyticsClient
         private readonly TextAnalyticsClientOptions _options;
         private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
         private const string AuthorizationHeader = "Ocp-Apim-Subscription-Key";
+        private IDictionary<string, int> _idToIndexMap;
```

Comment:
we can't do that. This can create a race condition if two methods with different inputs are called. the index map needs to be independent and pass from function to function